tests/iscsi/iscsi_client.pm

```diff
@@ -15,48 +15,130 @@ use base "x11test";
     wait_still_screen(2, 10);
-    type_string "iqn.2016-02.de.openqa";
+    type_string $initiator_conf{name};
     wait_still_screen(2, 10);
     assert_screen 'iscsi-initiator-service';
-    send_key "alt-v";             # go to discovered targets tab
+}
+
+sub initiator_discovered_targets_tab {
+    send_key "alt-v";    # go to discovered targets tab
     assert_screen 'iscsi-discovered-targets', 120;
-    send_key "alt-d";             # press discovery button
+    send_key "alt-d";    # press discovery button
     assert_screen 'iscsi-discovery';
-    send_key "alt-i";             # go to IP address field
+    send_key "alt-i";    # go to IP address field
     wait_still_screen(2, 10);
-    type_string "10.0.2.1";
+    type_string $initiator_conf{target_ip};
     assert_screen 'iscsi-initiator-discovered-IP-adress';
     send_key "alt-n";                                     # next
     assert_and_click 'iscsi-initiator-connect-button';    # press connect button
     assert_screen 'iscsi-initiator-connect-manual';
-    send_key "alt-n";                                     # go to connected targets tab
+    send_key 'alt-o';
+    assert_screen 'iscsi-initiator-discovery-enable-login-auth';
+    send_key 'alt-u';
+    type_string $initiator_conf{acl_initiator_user};
+    assert_screen 'iscsi-initiator-discovery-auth-initiators-username';
+    send_key 'alt-p';
+    my $init_pass = reverse $initiator_conf{acl_pass};
+    wait_screen_change { type_string $init_pass; };
+    send_key 'alt-r';
+    type_string $initiator_conf{acl_target_user};
+    assert_screen 'iscsi-initiator-discovery-auth-targets-username';
+    send_key 'alt-a';
+    wait_screen_change { type_string $initiator_conf{acl_pass}; };
+    send_key 'alt-n';
+}
+
+sub initiator_connected_targets_tab {
+    send_key "alt-d";    # go to discovered targets tab
     assert_screen 'iscsi-initiator-discovered-targets';
-    send_key "alt-n";                                     # next
+    send_key "alt-n";    # go to connected targets tab
     assert_screen 'iscsi-initiator-connected-targets';
-    send_key "alt-o";                                     # OK
-    wait_still_screen(2, 10);
+    send_key "alt-o";    # OK
+}
+
+
+sub run {
+    prepare_xterm;
+    record_info 'Network', 'Configure MM network - client';
+    configure_static_network($initiator_conf{ip});
+    mutex_wait('iscsi_target_ready', undef, 'Target configuration in progress!');
+    record_info 'Target Ready!', 'iSCSI target is configured, start initiator configuration';
+    type_string "yast2 iscsi-client; echo yast2-iscsi-client-\$? > /dev/$serialdev\n";
+    initiator_service_tab;
+    initiator_discovered_targets_tab;
+    initiator_connected_targets_tab;
+    wait_serial("yast2-iscsi-client-0", 180) || die "'yast2 iscsi-client ' didn't finish or exited with non-zero code";
+    # logging in to a target will create a local disc device
+    # it takes a moment, since udev actually handles it
+    sleep 5;
```

Comment:
This sleep same as the `sleep 2` and `sleep 3` further below seem rather arbitrary and not very robust. Would it work to try something in a loop? Is it the next command about `systemctl is-active iscsid.service` maybe?

Reply:
It is action triggered on target machine. I am not sure whether there is any chance how to check it from the initiator.

Reply:
Then maybe a mutex or barrier can help